### PR TITLE
Bug fix: Merge metadata logic for containerEnv - devcontainer build

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Fixes - https://github.com/devcontainers/cli/issues/381

### CODE REVIEW

**Comment by samruddhikhandale** [comment]
The following tests are failing https://github.com/devcontainers/cli/actions/runs/4069380934/jobs/7009165123#step:7:4476
since quite a few days and is not related to this PR changes

Fetching a manifest is failing with the following error 👇 

> Did not fetch manifest: {“errors”:[{“code”:“MANIFEST_UNKNOWN”,“message”:“OCI index found, but accept header does not support OCI indexes”}]}

Looking at how to fix it!

File: src/spec-node/containerFeatures.ts [hunk 1]
```diff
@@ -294,6 +294,14 @@ async function getFeaturesBuildOptions(params: DockerResolverParameters, devCont
 		.replace('#{copyFeatureBuildStages}', getCopyFeatureBuildStages(featuresConfig, buildStageScripts))
 		.replace('#{devcontainerMetadata}', getDevcontainerMetadataLabel(imageMetadata, common.experimentalImageMetadata))
 		;
+	
+	const rawContainerEnv = imageMetadata.raw.find(entry => entry.containerEnv)?.containerEnv;
```
Comment (chrmarti) [inline src/spec-node/containerFeatures.ts:298]: Use `imageMetadata.config`, that has variables like `${remoteWorkspaceFolder}` replaced. (`imageMetadata.raw` comes without these variables replaced.)

File: src/spec-node/containerFeatures.ts [hunk 2]
```diff
@@ -294,6 +294,14 @@ async function getFeaturesBuildOptions(params: DockerResolverParameters, devCont
...
+	if (rawContainerEnv !== undefined) {
+		let keys = Object.keys(rawContainerEnv);
+		const concatenatedEnv = keys.map(k => `ENV ${k}=${rawContainerEnv![k]}`).join('\n');
+		dockerfile += concatenatedEnv;
```
Comment (chrmarti) [inline src/spec-node/containerFeatures.ts:302]: Maybe add a placeholder in `getContainerFeaturesBaseDockerFile()` and add a `replace` line above instead to use the same pattern as above?

File: src/spec-node/containerFeatures.ts (hunk 1 above)
Comment (chrmarti) [reply src/spec-node/containerFeatures.ts:298]: `imageMetadata` actually includes the features for which we already apply `containerEnv`. Can you try using `devContainerConfig.config.containerEnv`?

**Review by chrmarti — CHANGES_REQUESTED** [review]
Makes sense. Should we remove `containerEnv` from `up` then?

**Comment by chrmarti** [comment]
I just now see: We still need `containerEnv` in `up` for the cases when we can avoid building a custom image (`getExtendImageBuildInfo` then skips `getFeaturesBuildOptions`).

File: src/spec-node/containerFeatures.ts (hunk 1 above)
Comment (samruddhikhandale) [reply src/spec-node/containerFeatures.ts]: > Can you try using devContainerConfig.config.containerEnv?

Works 👍

File: src/spec-node/containerFeatures.ts (hunk 2 above)
Comment (samruddhikhandale) [reply src/spec-node/containerFeatures.ts]: Updated

**Review by chrmarti — APPROVED** [review]
LGTM!

**Pull request merged**